web/src/unread_ops.ts

```diff
@@ -321,6 +351,18 @@ function bulk_update_read_flags_for_narrow(
     });
 }
 
+function handle_mark_messages_as_read(modal_id: string): void {
+    const selected_option = $("#mark_as_read_option").val();
+
+    if (selected_option === "muted_topics") {
+        mark_muted_topic_messages_as_read(modal_id);
+    } else if (selected_option === "topics_not_followed") {
+        mark_unfollowed_topic_messages_as_read(modal_id);
+    } else {
```

Comment:
Do an `else if` here and throw an exception if any other value arrives. It's good defensive programming to not assume your input is valid.